### PR TITLE
Sy 1390 add linting for type imports and arrows

### PR DESCRIPTION
**Pull request by pjdotson**
# Feature Pull Request Template

## Key Information

- **Linear Issue**: [SY-1390](https://linear.app/synnax/issue/)

## Description

Added new rules for linting to keep formatting consistent, including needing type in front of all imports imported as a type, and preferring arrow call backs.

## Basic Readiness

- [x] I have performed a self-review of my code.
- [x] I have added relevant tests to cover the changes to CI.
- [x] I have needed QA steps to the [release candidate](/.github/PULL_REQUEST_TEMPLATE/rc.md) template that cover these changes.
- [x] I have updated in-code documentation to reflect the changes.
- [x] I have updated user-facing documentation to reflect the changes.

## Backwards Compatibility

The following makes sure that this feature does not break backwards compatibility.

### Data Structures

- [x] Server - I have ensured that previous versions of stored data structures are properly migrated to new formats.
- [x] Console - I have ensured that previous versions of stored data structures are properly migrated to new formats.

### API Changes

- [x] Server - The server API is backwards-compatible
- The following client APIs are backwards-compatible:
  - [x] C++
  - [x] TypeScript
  - [x] Python

### Breaking Changes

If anything in this section is not true, please list all breaking changes.

### CODE REVIEW

File: console/src/group/ontology.tsx [hunk 1]
```diff
@@ -249,7 +249,7 @@ export const useCreateFromSelection = (): ((
       if (selection.parent == null) return;
       const resourcesToGroup = getResourcesToGroup(selection);
       const prevNodes = Tree.deepCopy(nodes);
-      let nextNodes = Tree.setNode({
+      Tree.setNode({
```
Comment (emilbon99) [inline console/src/group/ontology.tsx:252]: This function produces a shallow copy of the tree, which is why we had the assignment here. Can you add it back?

File: docs/site/src/components/console/OSDownloadButton.tsx [hunk 2]
```diff
@@ -7,11 +7,10 @@
 // License, use of this software will be governed by the Apache License, Version 2.0,
 // included in the file licenses/APL.txt.
 
-import { type ReactElement, useState, useEffect } from "react";
-
 import { Icon } from "@synnaxlabs/media";
 import { Button } from "@synnaxlabs/pluto";
 import { runtime } from "@synnaxlabs/x";
+import { type ReactElement, useEffect,useState } from "react";
```
Comment (emilbon99) [inline docs/site/src/components/console/OSDownloadButton.tsx:13]: This looks badly formatted

File: drift/examples/tauri/src/store.ts [hunk 3]
```diff
@@ -7,12 +7,11 @@
 // License, use of this software will be governed by the Apache License, Version 2.0,
 // included in the file licenses/APL.txt.
 
-import { createSlice, combineReducers } from "@reduxjs/toolkit";
-
+import { combineReducers,createSlice } from "@reduxjs/toolkit";
```
Comment (emilbon99) [inline drift/examples/tauri/src/store.ts:10]: Weird formatting

**Review by emilbon99 — APPROVED** [review]
There are some formatting issues and the build fails on windows. This pull requests will also cause merge conflicts with others, especially https://github.com/synnaxlabs/synnax/pull/912

**Comment by Lham42** [comment]
deploy console still failing

**Pull request merged**